src/platform/plugins/shared/unified_search/public/dataview_picker/change_dataview.tsx

```diff
@@ -99,16 +103,18 @@ export function ChangeDataView({
 
   useEffect(() => {
     const fetchDataViews = async () => {
-      const savedDataViewRefs = savedDataViews
+      const availableDataViewRefs = savedDataViews
         ? savedDataViews
         : (await data.dataViews.getIdsWithTitle()) ?? [];
+      const savedDataViewRefs = availableDataViewRefs.filter((dataView) => !dataView.managed);
+      const managedDataViewRefs =
+        savedDataViews?.filter((dataView) => dataView.managed === true).map(mapManagedDataView) ??
+        [];
       const adHocDataViewRefs = adHocDataViews?.map(mapAdHocDataView) ?? [];
-      const managedDataViewRefs = managedDataViews?.map(mapManagedDataView) ?? [];
-
       setDataViewsList([...savedDataViewRefs, ...adHocDataViewRefs, ...managedDataViewRefs]);
     };
     fetchDataViews();
-  }, [data, currentDataViewId, adHocDataViews, savedDataViews, managedDataViews]);
+  }, [data, currentDataViewId, adHocDataViews, savedDataViews]);
```

Comment:
why is `currentDataViewId` part of the deps list here?

Reply:
Reading from the [original PR](https://github.com/elastic/kibana/pull/126560/files), the data views will be refetch when the current id is changed

```
  useEffect(() => {
    const fetchDataViews = async () => {
      const dataViewsRefs = await data.dataViews.getIdsWithTitle();
      setDataViewsList(dataViewsRefs);
    };
    fetchDataViews();
  }, [data, currentDataViewId]);
```